get_child_pids.c

```diff
@@ -37,7 +37,7 @@ asmlinkage long sys_get_child_pids(pid_t* list, size_t limit, size_t* num_childr
 				}
 			return -ENOBUFS;
 			}
-			kfree(pPids);
+			//kfree(pPids);   //if you free here it won't be possible to be read
```

Comment:
It's fine because the data is copied back to user space before freeing the pointer.